tools/build.py

```diff
@@ -190,19 +196,22 @@ def adjust_options(options):
     # First fix some option inconsistencies
     if options.target_os == 'nuttx':
         options.buildlib = True
-        if not options.nuttx_home:
-            ex.fail('--nuttx-home needed for nuttx target')
-
-        options.nuttx_home = fs.abspath(options.nuttx_home)
-        if not fs.exists(options.nuttx_home):
-            ex.fail('--nuttx-home %s does not exists' % options.nuttx_home)
-
+        if not options.nuttx_home and not options.sysroot:
```

Comment:
If we want to keep the `--nuttx-home` option, but simplify this part we could specify the argument's target variable to be same as the `--sysroot` argument so then we would only have the `options.sysroot`.